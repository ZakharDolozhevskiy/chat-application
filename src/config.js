export const api = {
  host: 'http://185.13.90.140:8081/',
  channel: 'message'
};

export const routes = {
  MAIN: '/',
  SETTINGS: '/settings'
};