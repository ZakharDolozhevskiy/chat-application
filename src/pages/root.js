import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { Route } from 'react-router-dom';

import Chat from './chat';
import Header from '../components/header/index';
import Settings from './settings';
import { routes } from '../config';

export class Root extends React.PureComponent {
  render() {
    const { path, counter, className } = this.props;

    return (
      <section className={className}>
        <Header counter={counter} path={path}/>
        <Route path={routes.MAIN} component={Chat}/>
        <Route path={routes.SETTINGS} component={Settings}/>
      </section>
    );
  }
}

Root.propTypes = {
  path: PropTypes.string.isRequired,
  counter: PropTypes.number.isRequired,
  className: PropTypes.string
};

export default styled(Root)`
    display: flex;
    flex-direction: column;
    box-sizing: border-box;
    min-height: 100%;
    margin: 0 auto;
    color: ${p => p.theme.textColor};
    background-color: ${p => p.theme.backgroundColor};
`;