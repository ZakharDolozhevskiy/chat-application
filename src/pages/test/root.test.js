import React from 'react';
import toJson from 'enzyme-to-json';
import { Root } from '../root';
import { shallow } from 'enzyme';

import { routes } from '../../config';

jest.mock('react-router-dom', () => ({ Route: props => null }));

describe('<Root/>', () => {
  const component = shallow(
    <Root
      counter={0}
      path={routes.MAIN}
      translate={id => id}
      className=""
    />
  );

  it('should render root component', () => {
    expect(toJson(component)).toMatchSnapshot();
  });
});