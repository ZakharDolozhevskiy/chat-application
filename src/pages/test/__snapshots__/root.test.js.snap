// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Root/> should render root component 1`] = `
<section
  className=""
>
  <Styled(Header)
    counter={0}
    path="/"
  />
  <Route
    component={[Function]}
    path="/"
  />
  <Route
    component={[Function]}
    path="/settings"
  />
</section>
`;
